receiver/kafkareceiver/kafka_receiver.go

```diff
@@ -579,10 +593,29 @@ func (c *tracesConsumerGroupHandler) ConsumeClaim(session sarama.ConsumerGroupSe
 			err = c.nextConsumer.ConsumeTraces(session.Context(), traces)
 			c.obsrecv.EndTracesOp(ctx, c.unmarshaler.Encoding(), spanCount, err)
 			if err != nil {
-				if c.messageMarking.After && c.messageMarking.OnError {
-					session.MarkMessage(message, "")
+				if errorRequiresBackoff(err) && c.backOff != nil {
+					backOffDelay := c.backOff.NextBackOff()
+					if backOffDelay == backoff.Stop {
+						if c.messageMarking.After && c.messageMarking.OnError {
+							session.MarkMessage(message, "")
+						}
+						return err
+					}
+					select {
+					case <-session.Context().Done():
+						return nil
+					case <-time.After(backOffDelay):
+						if !c.messageMarking.After {
+							// Unmark the message so it can be retried
+							session.ResetOffset(claim.Topic(), claim.Partition(), message.Offset, "")
+						}
+					}
+				} else {
+					if c.messageMarking.After && c.messageMarking.OnError {
+						session.MarkMessage(message, "")
+					}
+					return err
```

Comment:
I'm a little bit confused that returning the error here will interrupt the for loop and return from `ConsumeClaim` function, how do we get back to `ConsumeClaim` again to consume the next messages?